specification/sections/02-providers.md

```diff
@@ -244,6 +244,30 @@ class MyProvider implements Provider, AutoDisposable {
   void dispose() {
     // close connections, terminate threads or timers, etc...
   }
+```
+
+### 2.6. Provider context reconciliation
+
+[![experimental](https://img.shields.io/static/v1?label=Status&message=experimental&color=orange)](https://github.com/open-feature/spec/tree/main/specification#experimental)
+
+Static-context focused providers may need a mechanism to understand when their cache of evaluated flags must be invalidated or updated. An `on-context-set` handler can be defined which performs whatever operations are needed to reconcile the evaluated flags with the new context.
+
+#### Requirement 2.6.1
+
+> The provider **MAY** define an `on context changed` handler, which takes an argument for the previous context and the newly set context, and reconciles any stored state pertaining to the global evaluation context.
+
+Especially in static-context implementations, providers and underlying SDKs may maintain a cache of evaluated flags for a particular context.
```

Comment:
why `local state` over `local cache`, is describing it as a `cache` too specific? I think describing it as a `cache` is likely easier to understand.

Reply:
I think I agree with @jonathannorris on this one. Because this is more of an example and not normative text, I think the use of "cache" is more concrete and helpful in communicating the idea.

Reply:
I think the "pertaining to flag evaluations" versus "evaluated flags" part may be more important. An SDK may have local state which it uses to perform evaluations, or it may have local state which is fully evaluated, maybe the state that changes isn't directly flags at all, but segmenting information, or metadata used to drive analytics.

Reply:
"maintain state for a particular context"

Though, it is probably also fine as is, because it isn't normative.

Reply:
:thinking:

Reply:
I've gone with `maintain state for a particular context` as @kinyoklion  recommends, which (I think) satisfies @moredip 's original concern and reads a bit better.